### PR TITLE
Add --all flag for bionemo_data_download

### PR DESCRIPTION
**Pull request by malcolmgreaves**
Adds a new flag, --all, to bionemo_data_download. This will override the other flags and download every possible resource, data and model, using the user-supplied --source.

This PR also refactors the script so that the variables are referenced to explicitly, rather than using the opaque argument parsed result object (args), aiding in readability & maintainability.

### CODE REVIEW

**Comment by pstjohn** [comment]
Why do we need this? I'd prefer not to have this option, only because IMO it will end up getting used in CI somewhere, and then we'll have to be very careful adding new resources (i.e., full ESM2 training data, ~100Gb) that might get picked up by accident in a download step.

**Comment by malcolmgreaves** [comment]
@pstjohn on the original PR before I split this out, there was consensus on a desire to have this very simple flag.

File: sub-packages/bionemo-testing/src/bionemo/testing/data/load.py [hunk 1]
```diff
@@ -220,30 +233,61 @@ def main_cli():
...
+                local_path = load(artifact_name, source=source)
+            # Print the result
+            print(str(local_path.absolute()))
+        else:
+            parser.error("You must provide an artifact name if --list-resources or --all is not set!")
+
+    # Any exceptions encountered above will cause the program to exist w/ a non-zero exit code.
+    # Additionally, the `parser.error` call will end the program with a non-zero exit code.
+
+    sys.exit(0)  # Successful exit
```
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply sub-packages/bionemo-testing/src/bionemo/testing/data/load.py:290]: Yes, this is not necessary. But this is a holdover from the original code.

In general, I will make changes to existing code in the least-disruptive way as possible. Sometimes that means keeping around things I don't necessarily agree with 🙃

**Review by tshimko-nv — APPROVED** [review]

**Review by dorotat-nv — APPROVED** [review]

File: sub-packages/bionemo-testing/src/bionemo/testing/data/load.py [hunk 2]
```diff
@@ -220,30 +233,61 @@ def main_cli():
...
-        # Print the result
-        print(str(local_path.absolute()))
+    download_all: bool = args.all
+    list_resources: bool = args.list_resources
+    artifact_name: str = args.artifact_name
+    source: Literal["pbss", "ngc"] = args.source
+
+    # main script logic
+    if download_all:
+        print("Downloading all resources:", file=sys.stderr)
```
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply sub-packages/bionemo-testing/src/bionemo/testing/data/load.py:256]: Yeah that'd be nice. We don't have any sort of centralized logging set up right now, so I'll keep this as a todo to come back and refactor when we add it in another PR.

File: sub-packages/bionemo-testing/src/bionemo/testing/data/load.py [hunk 3]
```diff
@@ -220,30 +233,61 @@ def main_cli():
...
+
+        print("-" * 80, file=sys.stderr)
+        print("All resources downloaded:", file=sys.stderr)
+        for resource_name, local_path in sorted(resource_to_local.items()):
+            print(f"  {resource_name}: {str(local_path.absolute())}", file=sys.stderr)
+
     else:
-        parser.error("You must provide an artifact name if --list-resources is not set.")
+        if list_resources:
+            print_resources(output_source=sys.stdin)
```
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply sub-packages/bionemo-testing/src/bionemo/testing/data/load.py:276]: Related to https://github.com/NVIDIA/bionemo-framework/pull/358#discussion_r1815358089 -- I'm just making as minor of modifications to the existing code as possible.

**Comment by malcolmgreaves** [comment]
@pstjohn I had some spare cycles and refactored a bit, taking in some of your feedback.

**Pull request merged**